llvm/lib/CodeGen/RegAllocGreedy.cpp

```diff
@@ -2375,10 +2375,12 @@ void RAGreedy::aboutToRemoveInterval(const LiveInterval &LI) {
 }
 
 void RAGreedy::initializeCSRCost() {
-  // We use the larger one out of the command-line option and the value report
-  // by TRI.
+  // We use the command-line option if it is explicitly set, otherwise use the
+  // larger one out of the command-line option and the value reported by TRI.
   CSRCost = BlockFrequency(
-      std::max((unsigned)CSRFirstTimeCost, TRI->getCSRFirstUseCost()));
+      CSRFirstTimeCost.getNumOccurrences()
+          ? CSRFirstTimeCost
+          : std::max((unsigned)CSRFirstTimeCost, TRI->getCSRFirstUseCost()));
```

Comment:
If the command line option isn't explicitly set, `CSRFirstTimeCost` will be the default value of 0 so this max will always return `TRI->getCSRFirstUseCost()`

Reply:
If someone goes in and changes the default value, wouldn't we want the comparison to occur without having to change the code here?

Reply:
I guess, but I'm not sure why the default of the `CSRFirstTimeCost` would be changed instead of the base class implementation of `TargetRegisterInfo::getCSRFirstUseCost()`.

We can take the code how you have it.